# Introdução à Ciência de Dados - UFPB
Professor: Yuri Malheiros

## Medidas de Centralidade e Dispersão

A partir do dataset `movies.csv` calcule a média, mediana e desvio padrão para as colunas:

### US Gross

In [217]:
# ADICIONE O SEU CÓDIGO AQUI
import pandas as pd
import numpy as np
import os
CURRENT_DIR=os.path.abspath(os.getcwd())
dataset = pd.read_csv(CURRENT_DIR+'/datasets/'+'movies.csv')

meanGross = np.mean(dataset['US Gross'])
medianGross = dataset['US Gross'].median()
standDeviation = np.sqrt(np.var(dataset['US Gross']))

print("|  Media: {}".format(meanGross))
print("|  Mediana: {}".format(medianGross))
print("|  Desvio padrão: {}".format(standDeviation))

|  Media: 44002085.163744524
|  Mediana: 22019465.5
|  Desvio padrão: 62545517.99547244


### Production Budget

In [219]:
# ADICIONE O SEU CÓDIGO AQUI

meanProduction = np.mean(dataset['Production Budget'])

medianProduction = dataset['Production Budget'].median()

standDeviation = np.sqrt(np.var(dataset['Production Budget']))


print("|  Media: {}".format(meanProduction))
print("|  Mediana: {}".format(medianProduction))
print("|  Desvio padrão: {}".format(standDeviation))

|  Media: 31069171.4484375
|  Mediana: 20000000.0
|  Desvio padrão: 35580352.71120245


### Running Time

In [221]:
# ADICIONE O SEU CÓDIGO AQUI

meanRunning = np.mean(dataset['Running Time (min)'])

medianRunning =dataset['Running Time (min)'].median()

standDeviation = np.sqrt(np.var(dataset['Running Time (min)']))


print("|  Media: {}".format(meanRunning))
print("|  Mediana: {}".format(medianRunning))
print("|  Desvio padrão: {}".format(standDeviation))

|  Media: 110.19354838709677
|  Mediana: 107.0
|  Desvio padrão: 20.16266992798399


### IMDB Rating

In [222]:
# ADICIONE O SEU CÓDIGO AQUI

meanRating = np.mean(dataset['IMDB Rating'])

medianRating =dataset['IMDB Rating'].median()

standDeviation = np.sqrt(np.var(dataset['IMDB Rating']))


print("|  Media: {}".format(meanRating))
print("|  Mediana: {}".format(medianRating))
print("|  Desvio padrão: {}".format(standDeviation))

|  Media: 6.283467202141901
|  Mediana: 6.4
|  Desvio padrão: 1.2520803678620684


### IMDB Votes

In [223]:
# ADICIONE O SEU CÓDIGO AQUI

meanVotes = np.mean(dataset['IMDB Votes'])

medianVotes =dataset['IMDB Votes'].median()

standDeviation = np.sqrt(np.var(dataset['IMDB Votes']))


print("|  Media: {}".format(meanVotes))
print("|  Mediana: {}".format(medianVotes))
print("|  Desvio padrão: {}".format(standDeviation))

|  Media: 29908.64457831325
|  Mediana: 15106.0
|  Desvio padrão: 44930.06203002918


### Análise dos Resultados

Quais colunas tiveram valores muito diferentes entre média e mediana? Quais os seus desvios padrão?

**RESPONDA AQUI**

O US Gross, obteve uma alta descrepância entre a média e seu desvio padrão
|  Desvio padrão: 62545517.99547244

Production Budget, obteve uma alta descrepância entre a média e seu desvio padrão
|  Desvio padrão: 35580352.71120245

IMDB Votes, obteve uma alta descrepância entre a média e seu desvio padrão
|  Desvio padrão: 44930.06203002918

Que conclusões podemos tirar desses resultados?

**RESPONDA AQUI**

Não podemos ter um bom resultado ou extrair boas informações dos dados  que contém maior vâriancia pois a média não é um bom representativo para os dados analisados.
Já nos casos que possuiram poucam descrepância entre o seu desvio padrão, podemos extrair uma boa estimativa média dos seus dados como Running Time e IMDB Rating, esses ajudariam uma avaliação dos proximos filmes para uma predição de qual seu tempo de duração e sua avaliação.